src/test/java/org/opensearch/knn/index/FaissIT.java

```diff
@@ -151,6 +151,181 @@ public void testEndToEnd_fromMethod() throws Exception {
+            .startObject("properties")
+            .startObject(fieldName)
+            .field("type", "knn_vector")
+            .field("dimension", dimension)
+            .startObject(KNNConstants.KNN_METHOD)
+            .field(KNNConstants.NAME, nsgMethod.getMethodComponent().getName())
+            .field(KNNConstants.METHOD_PARAMETER_SPACE_TYPE, spaceType.getValue())
+            .field(KNNConstants.KNN_ENGINE, KNNEngine.FAISS.getName())
+                .startObject(KNNConstants.PARAMETERS)
+                .field(KNNConstants.METHOD_PARAMETER_M, 64)
+                .endObject()
+            .endObject()
+            .endObject()
+            .endObject()
+            .endObject();
+
+        Map<String, Object> mappingMap = xContentBuilderToMap(builder);
+        String mapping = Strings.toString(builder);
+
+        createKnnIndex(indexName, mapping);
+        Map<String, Object>  maping = getIndexMappingAsMap(indexName);
+//        assertEquals(new TreeMap<>(mappingMap), new TreeMap<>(maping));
+
+        // Index the test data
+        bulkAddKnnDocs(indexName, fieldName, testData.indexData.vectors, testData.indexData.docs.length);
+//        for (int i = 0; i < testData.indexData.docs.length; i++) {
+//            addKnnDoc(
+//                indexName,
+//                Integer.toString(testData.indexData.docs[i]),
+//                fieldName,
+//                Floats.asList(testData.indexData.vectors[i]).toArray()
+//            );
+//        }
+
+        // Assert we have the right number of documents in the index
+        refreshAllNonSystemIndices();
+        assertEquals(testData.indexData.docs.length, getDocCount(indexName));
+
+        int k = 10;
+        for (int i = 0; i < testData.queries.length; i++) {
+            Response response = searchKNNIndex(indexName, new KNNQueryBuilder(fieldName, testData.queries[i], k), k);
+            String responseBody = EntityUtils.toString(response.getEntity());
+            List<KNNResult> knnResults = parseSearchResponse(responseBody, fieldName);
+            assertEquals(k, knnResults.size());
+
+            List<Float> actualScores = parseSearchResponseScore(responseBody, fieldName);
+            for (int j = 0; j < k; j++) {
+                float[] primitiveArray = Floats.toArray(Arrays.stream(knnResults.get(j).getVector()).collect(Collectors.toList()));
+                assertEquals(
+                    KNNEngine.FAISS.score(KNNScoringUtil.l2Squared(testData.queries[i], primitiveArray), spaceType),
+                    actualScores.get(j),
+                    0.0001
+                );
+            }
+        }
+
+        // Delete index
+        deleteKNNIndex(indexName);
+
+        // Search every 5 seconds 14 times to confirm graph gets evicted
+        int intervals = 14;
+        for (int i = 0; i < intervals; i++) {
+            if (getTotalGraphsInCache() == 0) {
+                return;
+            }
+
+            Thread.sleep(5 * 1000);
+        }
```

Comment:
you can move this to base class so that other tests can also use this, or this can be part of deleteIndex function. I think making it part of deleteKNNIndex function is better choice.

Also, why we are not putting these delete in a function annotated with @After

Reply:
i just copy these code from `NmslibIT#testEndToEnd` and `FaissIT#testEndToEnd_fromMethod` and `OpenSearchIT#testEndToEnd` 

> Also, why we are not putting these delete in a function annotated with @after

i think it is great to move the same logic into  annotated with @after.

and these need modified many tests, so i think we can create a new PR for fix it

Reply:
Make sense. We can create a separate PR for that.